src/views/virtualmachines/details/tabs/metrics/MigrationCharts/MigrationProgressStatus.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+
+import { V1VirtualMachineInstance } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import useMigrationPercentage from '@kubevirt-utils/resources/vm/hooks/useMigrationPercentage';
+import { Timestamp } from '@openshift-console/dynamic-plugin-sdk';
+import {
+  Card,
+  CardBody,
+  CardTitle,
+  Progress,
+  ProgressMeasureLocation,
+  ProgressVariant,
+  Stack,
+  StackItem,
+} from '@patternfly/react-core';
+
+type MigrationProgressStatusProps = {
+  vmi: V1VirtualMachineInstance;
+};
+
+const MigrationProgressStatus: React.FC<MigrationProgressStatusProps> = ({ vmi }) => {
+  const { t } = useKubevirtTranslation();
+
+  const { endTimestamp, isFailed, percentage } = useMigrationPercentage(vmi);
+
+  let progressStatus = null;
+
+  if (percentage === 100) progressStatus = ProgressVariant.success;
+
+  if (isFailed) progressStatus = ProgressVariant.danger;
```

Comment:
why u need let? percentage is reactive which means it will update all the rest of file if the value changes.

const progressStatus = isFailed ? ProgressVariant.danger : percentage === 100 && ProgressVariant.success

or as a function

getVariant = () => {
if (percentage === 100) return  ProgressVariant.success;
if (isFailed) return ProgressVariant.danger;

return ProgressVariant.x
}

can be a case of percentage = X (which is not 100) ?